windows-msi/version.m4

```diff
@@ -0,0 +1,37 @@
+dnl ============================================================
+dnl Downloadables
+dnl ============================================================
+
+dnl TAP-Windows binaries (URL to .zip file containing "tap-windows-[PRODUCT_TAP_WIN_VERSION]" folder with driver)
+define([PRODUCT_TAP_WIN_VERSION],      [9.21.2])
+define([PRODUCT_TAP_WIN_URL],          [http://build.openvpn.net/downloads/releases/tap-windows-9.21.2.zip])
+define([PRODUCT_TAP_WIN_COMPONENT_ID], [tap0901])
+
+dnl OpenVPNServ2.exe binary
+define([OPENVPNSERV2_URL], [http://build.openvpn.net/downloads/releases/openvpnserv2-1.3.0.0.exe])
+
+dnl Easy RSA binaries (URL to .tar.gz file containing "easy-rsa-[EASYRSA_VERSION]" folder with Easy RSA)
+define([EASYRSA_VERSION], [2.3.3_master])
+define([EASYRSA_URL],     [http://build.openvpn.net/downloads/releases/easy-rsa-2.3.3_master.tar.gz])
+
+
+dnl ============================================================
+dnl MSI Provisioning
+dnl ============================================================
+
+dnl Define the product version
+define([PRODUCT_NAME],      [OpenVPN])
+define([PRODUCT_PUBLISHER], [OpenVPN Technologies, Inc.])
+define([PRODUCT_VERSION],   [2.4.6])
+define([PRODUCT_TAP_NAME],  [TAP-Windows])
+
+dnl The version GUID MUST change for each release.
+define([PRODUCT_VERSION_GUID], [{80493CF5-3B4C-4B46-9054-BFC3D6E13C0E}])
+
+dnl The upgrade GUIDs MUST persist for all versions of the same product line.
+dnl Please use own upgrade GUIDs when deploying a non-official OpenVPN release.
+define([PRODUCT_UPGRADE_GUID_x86],   [{1195A47B-A37A-4055-9D34-B7A691F7E97B}])
```

Comment:
How were/are these GUIDs created?

Reply:
`uuidgen | tr [a-f] [A-F]`

I just picked some random to get started. Any unique hexadecimal number is OK. Old MSI versions required all uppercase GUIDs and I stick to that recommendation.

`PRODUCT_UPGRADE_GUID_...` are called upgrade code(s) and are used by MSI to follow the product line across versions, 3rd party products to detect OpenVPN is installed... The upgrade code is rarely changed - it is _the_ product identifier. However, I choose not to hard-code them to allow 3rd party vendors to package their own "OpenVPN" product line.

Use of platform-specific upgrade codes teaches Group Policy Management Editor not to mix 32 and 64-bit packages when updating. Example: MS Office 2016 has something like `{00160000-008C-0000-0000-0000000FF1CE}` for 32-bit version and `{00160000-008C-0000-1000-0000000FF1CE}` for 64-bit.

If you want some fancy ones to mark the official line of 32 and 64-bit OpenVPN MSI packages, it is time to set them now.